pkg/pool-stable-phantom/contracts/test/MockStablePhantomPool.sol

```diff
@@ -15,14 +15,46 @@
 pragma solidity ^0.7.0;
 pragma experimental ABIEncoderV2;
 
+import "@balancer-labs/v2-pool-utils/contracts/test/MockInvariantDependency.sol";
+
 import "../StablePhantomPool.sol";
 
-contract MockStablePhantomPool is StablePhantomPool {
+contract MockStablePhantomPool is StablePhantomPool, MockInvariantDependency {
     constructor(NewPoolParams memory params) StablePhantomPool(params) {
         // solhint-disable-previous-line no-empty-blocks
     }
 
     function mockCacheTokenRateIfNecessary(IERC20 token) external {
         _cacheTokenRateIfNecessary(token);
     }
+
+    function getRate() public view virtual override whenInvariantConverges returns (uint256) {
+        return super.getRate();
+    }
```

Comment:
Let's take this a step further and instead of overriding `getRate` let's make `_scalingFactors()` revert.

Reply:
Maybe that should be an alternative thing though, as the scaling factors would revert if a dependency is broken, not the pool. Perhaps `InvariantDependency` can be renamed to e.g. `FailureModes`?

Reply:
Good idea - will generalize to rate providers failing